packages/trpc/server/routers/viewer/slots.ts

```diff
@@ -326,10 +368,31 @@ export async function getSchedule(input: z.infer<typeof getScheduleSchema>, ctx:
   }
 
   let availableTimeSlots: typeof timeSlots = [];
+  // Load cached busy slots
+  const selectedSlots = await prisma.selectedSlots.findMany({
+    where: {
+      eventTypeId: { equals: eventType.id },
+      releaseAt: { gt: dayjs.utc().format() },
+    },
+    select: { id: true, slotUtcDate: true },
+  });
+  const ids: number[] = [];
+  const slotBusy =
+    selectedSlots?.length > 0
+      ? selectedSlots.map((item) => {
+          ids.push(item.id);
+          return item.slotUtcDate.toISOString();
+        })
+      : [];
```

Comment:
No need to check length for an empty array. and findMany will always return an array. Not undefined.

```suggestion
  const slotBusy = selectedSlots.map((item) => {
    ids.push(item.id);
    return item.slotUtcDate.toISOString();
  });
```